src/v/transform/logging/probes.cc

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#include "transform/logging/probes.h"
+
+#include "base/seastarx.h"
+#include "config/configuration.h"
+#include "prometheus/prometheus_sanitize.h"
+
+#include <seastar/core/metrics_registration.hh>
+
+namespace transform::logging {
+
+void logger_probe::setup_metrics(model::transform_name_view transform_name) {
+    namespace sm = ss::metrics;
+
+    const auto name_label = sm::label("transform_name");
+    const std::vector<sm::label_instance> labels = {
+      name_label(transform_name()),
+    };
+
+    auto setup_common =
+      [this, &labels](const std::vector<sm::label>& aggregate_labels) {
+          std::vector<sm::metric_definition> defs;
+          defs.emplace_back(
+            sm::make_counter(
+              "events_total",
+              [this]() { return _total_log_events; },
+              sm::description("Running count of transform log events"),
+              labels)
+              .aggregate(aggregate_labels));
+          defs.emplace_back(
+            sm::make_counter(
+              "events_dropped_total",
+              [this]() { return _total_dropped_log_events; },
+              sm::description("Running count of dropped transform log events"),
+              labels)
+              .aggregate(aggregate_labels));
+          return defs;
+      };
+
+    auto group_name = prometheus_sanitize::metrics_name(
+      "data_transforms_logger");
+
+    if (!config::shard_local_cfg().disable_metrics()) {
```

Comment:
Do we really need all this song and dance? If so we really should clean this up and force consistency..(nothing you need to do)

Reply:
Yeah, I mean this is the "standard" pattern I think. Very tedious